tests/find_or_insert_test.cc.cu

```diff
@@ -0,0 +1,2309 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* 
+ * test APIs: find_or_insert and assign,
+ * move insert(mi) from `insert_or_assign` to `find`.
+ */
+
+#include <gtest/gtest.h>
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <algorithm>
+#include <array>
+#include <iostream>
+#include <thread>
+#include <vector>
+#include "merlin_hashtable.cuh"
+#include "test_util.cuh"
+
+constexpr size_t DIM = 16;
+using K = uint64_t;
+using V = float;
+using M = uint64_t;
+using Table = nv::merlin::HashTable<K, V, M>;
+using TableOptions = nv::merlin::HashTableOptions;
+
+template <class K, class M>
+__forceinline__ __device__ bool erase_if_pred(const K& key, M& meta,
+                                              const K& pattern,
+                                              const M& threshold) {
+  return ((key & 0x7f > pattern) && (meta > threshold));
+}
+
+template <class K, class M>
+__device__ Table::Pred EraseIfPred = erase_if_pred<K, M>;
+
+template <class K, class M>
+__forceinline__ __device__ bool export_if_pred(const K& key, M& meta,
+                                               const K& pattern,
+                                               const M& threshold) {
+  return meta > threshold;
+}
+
+template <class K, class M>
+__device__ Table::Pred ExportIfPred = export_if_pred<K, M>;
+
+
+void test_basic_mi(size_t max_hbm_for_vectors, bool use_constant_memory) {
```

Comment:
If everything has `mi`, the `mi` would be unnecessary. And now you make a seperate test file.

Reply:
OK, I get